### PR TITLE
Add & remove Functions

### PR DESCRIPTION
**Pull request by LYANGEND**
**I have Achieved he following milestones in this project**;

- Implement a function for adding a new task (add a new element to the array).

- Implement a function for deleting a task (remove an element from the array).

- Implement a function for editing task descriptions
- When a task is deleted all remaining items' indexes are updated, so they represent the current list order and are unique.

- All changes to the To-Do List are saved in local storage.

### CODE REVIEW

File: modules/addTask.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const addTask = (text) => {
+  let tasks = getTaskFromLocalStorage();
+  tasks = tasks.sort((v1, v2) => v1.index - v2.index);
+  let index;
+  if (tasks.length === 0) index = 1;
+  else index = tasks[tasks.length - 1].index + 1;
+  const obj = { index, description: text, completed: false };
+  tasks.push(obj);
+  setTaskIntoLocalStorage(tasks);
+};
```
Comment (ShoiraTa) [inline modules/addTask.js:12]: - [ ] Please make sure that deleting a task updates all remaining items' indexes, so they represent the current list order and are unique. For example you had task1 with index 1, task2  with index 2, after you delete task1, task2 should get index 1
so you will always have index starting from 1...

https://user-images.githubusercontent.com/77038610/160114243-4624b9b6-b35c-4faa-ad67-8be3db1921f2.mp4

File: modules/updateTask.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  const selectedID = parseInt(e.id, 10);
+
+  for (let i = 0; i < tasks.length; i += 1) {
+    if (tasks[i].index === selectedID) {
+      tasks[i].description = e.value;
+      break;
+    }
+  }
+  setTaskIntoLocalStorage(tasks);
+};
```
Comment (ShoiraTa) [inline modules/updateTask.js:14]: - [ ] I cannot edit a task, if I click on task description, bin icon gets aligned to the left and task description disappears making it impossible to update it. Please make sure that users can update the task description:

https://user-images.githubusercontent.com/77038610/160114179-d4c20291-e251-47e8-9bda-8f64153184f9.mp4

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @LYANGEND ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — COMMENTED** [review]
Hello @LYANGEND  👋,

_This is an additional review as a result of the second opinion request._

- As pointed out by the previous reviewer it is a requirement to update all remaining items' indexes when you delete a task so they represent the current list order and are unique. HINT 💭 : you can reset the array index after every delete i suggest you use the `filter ()` method which doesn't modify the original array after which you can reset the array index ( ie `const resetIndexes = (arr) => arr.forEach((item, idx) => { item.index = idx + 1; })`;) 
- As for the editing functionality it works perfectly fine 👍 

As this was the only reason that your project was not approved kindly make the necessary change and it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will review accordingly 👍 

Keep rocking 🚀!

**Review by Gambit142 — APPROVED** [review]
Hi @LYANGEND 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

#### It would be nice to apply the same logic of resetting the indexes in the array of tasks when you clear all completed tasks in your `completedTask` function. Please remove all `console.log` statements in your files as it is not good JavaScript practice to leave them in your codebase after debugging.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**